lib/web/imageRouter/imgur.js

```diff
@@ -15,12 +15,25 @@ exports.uploadImage = function (imagePath, callback) {
     return
   }
 
-  imgur.setClientId(config.imgur.clientID)
-  imgur.uploadFile(imagePath)
-    .then(function (json) {
+  const buffer = fs.readFileSync(imagePath)
+
+  const params = new URLSearchParams()
+  params.append('image', buffer.toString('base64'))
+  params.append('type', 'base64')
+  fetch('https://api.imgur.com/3/image', {
+    method: 'POST',
+    body: params,
+    headers: { Authorization: `Client-ID ${config.imgur.clientID}` }
```

Comment:
@DerMolly 

What if you don't provide an client id? The imgur lib defined a fallback id for this purpose. Maybe we should also fallback to it.

Reply:
@SISheogorath Does the current imgur support uploads without configuring a client id?

Reply:
If so we should probably continue to do so…

Reply:
If you inspect the code.. yes

setClientID accepts only strings. If you don't set the ID in your config, then the lib will continue to use the build-in ID.

Reply:
https://github.com/hackmdio/node-imgur/blob/de0a7a1f1eb2cb6628385fedb990ad396a190573/lib/imgur.js#L285

Reply:
Do we want to continue to use that ID or do we want our own? And do we want to continue this behavior in 2.0 (currently it fails with no clientId)?

Reply:
I think without a client ID, deletion of uploads is impossible?
If so, we should not allow an empty client id.

Reply:
To require a client id now would be a breaking change. Therefore we should keep a fallback id for now for 1.x

Reply:
I restored the default behavior for this PR